iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/LogRetentionCheck.java

```diff
@@ -68,6 +68,9 @@ protected void registerResourceConsumer() {
 
   private Consumer<ContextualResource> checkLogRetention(String propertyName, String enablingTypeName, String retentionDayTypeName) {
     return resource -> {
+      if (resource.isReferencingResource()) {
```

Comment:
I have mixed feelings about this condition in multiple checks. However, I understand that it's not possible to use `reportIfAbsent` with an embedded existence check. I have just one idea, however as I'm not sure about it myself feel free do discard it. The idea is to add a wrapper `skippingReferencingResources(Consumer<> resourceHandler)` and use it directly in `register` methods, like
```java
register("Microsoft.Network/firewallPolicies",
  skippingReferencingResources(checkLogRetention("insights", "isEnabled", RETENTION_DAY_NAME)));
```
This might give some improvements with respect to duplication and readability. WDYT?

Reply:
Good idea!